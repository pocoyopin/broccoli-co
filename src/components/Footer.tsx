import React from "react";
import styled from "styled-components";

const Footer: React.FC = () => {
  return (
    <Container>
      <Text>Made with &#x2661; in Singapore</Text>
      <Text>&#169; 2024 Broccoli & Co. All rights reserved.</Text>
    </Container>
  );
};

export default Footer;

const Container = styled.div`
  min-height: 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${({ theme }) => theme.spacing.d}px;
  border-top: 1px solid ${({ theme }) => theme.colors.black};
`;

const Text = styled.p`
  margin: ${({ theme }) => theme.spacing.b}px;
`;
