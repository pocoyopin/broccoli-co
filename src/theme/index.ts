const breakpoints = ["450px", "600px", "800px", "1200px"];

const colors = {
  black: "#000000",
  black10: "rgba(0,0,0,0.1)",
  black50: "rgba(0,0,0,0.5)",
  black70: "rgba(0,0,0,0.7)",
  black90: "rgba(0,0,0,0.9)",
  white: "#ffffff",
  red: "#ff0000",
};

const fontWeights = {
  light: 300,
  normal: 400,
  medium: 500,
  lightBold: 600,
  bold: 700,
};

const fontSizes = {
  small: "0.8rem",
  subtext: "1rem",
  paragraph: "1.2rem",
  h6: "1.35rem",
  h5: "1.5rem",
  h4: "1.8rem",
  h3: "2.4rem",
  h2: "3.2rem",
  h1: "4.8rem",
};

const spacing = {
  a: 0,
  b: 4,
  c: 8,
  d: 16,
  e: 24,
  f: 32,
};

const theme = {
  breakpoints,
  mediaQueries: {
    sm: `@media screen and (min-width: ${breakpoints[0]})`,
    md: `@media screen and (min-width: ${breakpoints[1]})`,
    lg: `@media screen and (min-width: ${breakpoints[2]})`,
    xl: `@media screen and (min-width: ${breakpoints[3]})`,
  },
  colors,
  fontWeights,
  fontSizes,
  spacing,
};

export default theme;
