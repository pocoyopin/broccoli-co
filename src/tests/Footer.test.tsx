import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import renderer from "react-test-renderer";

import theme from "../theme";
import Footer from "../components/Footer";

jest.mock("axios", () => {
  return { post: jest.fn() };
});

describe("Footer", () => {
  test("renders footer", () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>
    );

    renderer.create(component).toJSON();
    render(component);
    expect(screen.getByText("Made with ♡ in Singapore")).toBeInTheDocument();
    expect(
      screen.getByText("© 2024 Broccoli & Co. All rights reserved.")
    ).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
});
