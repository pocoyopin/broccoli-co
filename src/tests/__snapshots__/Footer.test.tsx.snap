// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer 1`] = `
Object {
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": Object {},
  "key": null,
  "props": Object {
    "children": Object {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": Object {},
      "key": null,
      "props": Object {},
      "type": [Function],
    },
    "theme": Object {
      "breakpoints": Array [
        "450px",
        "600px",
        "800px",
        "1200px",
      ],
      "colors": Object {
        "black": "#000000",
        "black10": "rgba(0,0,0,0.1)",
        "black50": "rgba(0,0,0,0.5)",
        "black70": "rgba(0,0,0,0.7)",
        "black90": "rgba(0,0,0,0.9)",
        "red": "#ff0000",
        "white": "#ffffff",
      },
      "fontSizes": Object {
        "h1": "4.8rem",
        "h2": "3.2rem",
        "h3": "2.4rem",
        "h4": "1.8rem",
        "h5": "1.5rem",
        "h6": "1.35rem",
        "paragraph": "1.2rem",
        "small": "0.8rem",
        "subtext": "1rem",
      },
      "fontWeights": Object {
        "bold": 700,
        "light": 300,
        "lightBold": 600,
        "medium": 500,
        "normal": 400,
      },
      "mediaQueries": Object {
        "lg": "@media screen and (min-width: 800px)",
        "md": "@media screen and (min-width: 600px)",
        "sm": "@media screen and (min-width: 450px)",
        "xl": "@media screen and (min-width: 1200px)",
      },
      "spacing": Object {
        "a": 0,
        "b": 4,
        "c": 8,
        "d": 16,
        "e": 24,
        "f": 32,
      },
    },
  },
  "type": [Function],
}
`;
